docs/reference/ilm/policy-definitions.asciidoc

```diff
@@ -0,0 +1,293 @@
+
+This example rolls the index over when it is at least 100 gigabytes.
+
+[source,js]
+--------------------------------------------------
+{
+  "rollover" : {
+    "max_size": "100GB"
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+==== Example: Rollover when index has too many documents
+
+This example rolls the index over when it contains at least
+1000000 documents.
+
+[source,js]
+--------------------------------------------------
+{
+  "rollover" : {
+    "max_docs": 1000000
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+==== Example: Rollover when index is too old
+
+This example rolls the index over when it has been created at least
+7 days ago.
+
+[source,js]
+--------------------------------------------------
+{
+  "rollover" : {
+    "max_age": "7d"
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+==== Example: Rollover when index is too old or too large
+
+This example rolls the index over when it has been created at least
+7 days ago or it is at least 100 gigabytes. In this case, the index will be rolled
+over when at least one of these two conditions is met.
+
+[source,js]
+--------------------------------------------------
+{
+  "rollover" : {
+    "max_age": "7d",
+    "max_size": "100GB"
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+
+[[ilm-shrink-action]]
+=== Shrink
+
+WARNING: This will make the index read-only (see: <<dynamic-index-settings,index.blocks.write>>)
+
+This action shrinks an existing index into a new index with fewer primary shards. It
+calls the <<indices-shrink-index,Shrink API>> to shrink the index. Since allocating
+all the primary shards of the index to one node is a prerequisite, this action will
+first allocate the primary shards to a valid node. After shrinking, it will swap aliases
+pointing to the original index into the new shrunken index. The new index will also
+have a new name: "shrink-<origin-index-name>". So if the original index was called "logs",
+then the new index will be named "shrink-logs".
+
+[[ilm-shrink-options]]
+.Shrink Options
+[options="header"]
+|======
+| Name               | Required  | Default             | Description
+| `number_of_shards` | yes       | -                   | The number of shards to shrink to. must be a factor of the number of shards in the source index.
+|======
+
+[source,js]
+--------------------------------------------------
+{
+  "shrink" : {
+    "number_of_shards": 1
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+== Phase Definitions
+
+Here are the phases and which actions can be defined within them.
+
+* Hot
+  - <<ilm-rollover-action,Rollover>>
+* Warm
+  - <<ilm-allocate-action,Allocate>>
+  - <<ilm-readonly-action,Read-Only>>
+  - <<ilm-forcemerge-action,Force Merge>>
+  - <<ilm-shrink-action,Shrink>>
+* Cold
+  - <<ilm-allocate-action,Allocate>>
+* Delete
+  - <<ilm-delete-action,Delete>>
```

Comment:
Is here a good place to go into details about the order of execution and conditions on 
execution?

for example:

```
The order in which phases are defined in a policy definition
can be arbitrary, but the order in which they are executed is
pre-determined by {ILM}. The order of execution is: Hot, Warm, Cold, Delete.

Phases can optionally set a `min_age` condition, which determines what
age an index must be before entering the associated phase.
```